support-utils/cre_astro_software_gather_logs.sh

```diff
@@ -0,0 +1,392 @@
+      echo "Exporting Release name ";export Release_Name=$(echo $NS| cut -c $Y-)
+      echo "Your Release_Name in current namespace $NS is $Release_Name."
+echo "======================Gathering All the Deployment namespace logs in the $NS======================"
+      echo "Gathering logs of scheduler in $NS Namespace ";kubectl logs deployment/$Release_Name-scheduler -c scheduler > "$NS/AllPodlogs/morelogs/scheduler_$NS.log" -n $NS  
+      echo "Gathering logs of worker in $NS Namespace ";kubectl logs deployment/$Release_Name-worker -c worker > "$NS/AllPodlogs/morelogs/worker_$NS.log" -n $NS
+      echo "Gathering logs of webserverin $NS Namespace ";kubectl logs deployment/$Release_Name-webserver -c webserver > "$NS/AllPodlogs/morelogs/webserver$NS.log" -n $NS
+      echo "Gathering logs of triggerer in $NS Namespace ";kubectl logs deployment/$Release_Name-triggerer -c triggerer > "$NS/AllPodlogs/morelogs/triggerer $NS.log" -n $NS
+      echo "Gathering logs of pgbouncer in $NS Namespace ";kubectl logs deployment/$Release_Name-pgbouncer -c pgbouncer > "$NS/AllPodlogs/morelogs/pgbouncer _$NS.log" -n $NS    
+      echo "Gathering logs of flower  in $NS Namespace ";kubectl logs deployment/$Release_Name-flower > "$NS/AllPodlogs/morelogs/flower _$NS.log" -n $NS
+      echo "Gathering logs of statsd in $NS Namespace ";kubectl logs deployment/$Release_Name-statsd > "$NS/AllPodlogs/morelogs/statsd_$NS.log" -n $NS
+      echo "Gathering logs of redis in $NS Namespace ";kubectl logs sts/$Release_Name-redis > "$NS/AllPodlogs/morelogs/redis_$NS.log" -n $NS 
+      echo "Gathering helm history in $NS Namespace";helm history $Release_Name -n $NS > "$NS/AllPodlogs/morelogs/helm_history_$Release_Name.yaml"
+      echo "Gathering helm values from $NS Namespace";helm get values $Release_Name -o yaml -n $NS  > "$NS/AllPodlogs/morelogs/helm_values_$Release_Name.yaml"
+    done
+
+
+  
+
+
+
+echo "Checking ENDPOINTS"
+
+for EP in $(kubectl describe svc kube-dns -n kube-system|grep Endpoints|awk '{print $2}'|uniq|sed 's/,/\n/g'|sed 's/:[^[:blank:]]*//'); do
+echo "======================CHECKING Houston ENDPOINT for $EP======================";nslookup houston.$BASEDOMAIN > $DIR/nslookup_houston_$EP.$BASEDOMAIN.log; echo " you have to run nslookup houston.$BASEDOMAIN $EP inside any of the pods lets say inside a nginx pod" >> $DIR/nslookup_houston_$EP.$BASEDOMAIN.log
+echo "PLEASE NOTE ======== you have to run nslookup houston.$BASEDOMAIN $EP inside any of the pods lets say inside a nginx pod to make sure endpoints are running fine"
+done
+
+for i in {1..10} ;do curl -I  https://registry.$BASEDOMAIN; done  > $DIR/curl_check_registry.$BASEDOMAIN.log
+for i in {1..10} ;do curl -I  https://app.$BASEDOMAIN/; done      > $DIR/curl_check_app.$BASEDOMAIN.log
+for i in {1..10} ;do curl -I  https://install.$BASEDOMAIN/; done  > $DIR/curl_check_install.$BASEDOMAIN.log
```

Comment:
What are we trying to do here? It seems like a loop is not the right approach.

Reply:
The intention was to test the responsiveness of the endpoints by sending multiple requests